### PR TITLE
feat: Add option to not sign message

### PR DESCRIPTION
**Pull request by Elyniss**

### CODE REVIEW

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/client/mapper/ClientMapper.kt [hunk 1]
```diff
@@ -77,7 +77,7 @@ internal fun SDKError.toClientError(): Chat.Model.Error = Chat.Model.Error(this.
 internal fun ConnectionState.toClient(): Chat.Model.ConnectionState = Chat.Model.ConnectionState(isAvailable)
 
 @JvmSynthetic
-internal fun Chat.Model.Cacao.Signature.toCommon(): Cacao.Signature = Cacao.Signature(t, s, m)
+internal fun Chat.Model.Cacao.Signature?.toCommon(): Cacao.Signature? = this?.let { cacao -> Cacao.Signature(cacao.t, cacao.s, cacao.m) }
```
Comment (jakubuid) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/client/mapper/ClientMapper.kt:80]: <3

**Review by jakubuid — APPROVED** [review]

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt [hunk 2]
```diff
@@ -235,7 +235,7 @@ internal class ChatEngine(
                 )
 
                 val message = payload.toCAIP122Message()
-                val signature = onSign(message)
+                val signature = onSign(message) ?: return
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt:238]: same comment here

**Review by TalhaAli00 — APPROVED** [review]

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt [hunk 3]
```diff
@@ -174,7 +174,7 @@ internal class ChatEngine(
             )
 
             val message = payload.toCAIP122Message()
-            val signature = onSign(message)
+            val signature = onSign(message) ?: return
```
[the remark this answers is not in the document]
Comment (Elyniss) [reply chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt:177]: I think we should silently return. Devs have to return null themselves. Calling onError will only confuse them, imo

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt (hunk 2 above)
Comment (Elyniss) [reply chat/sdk/src/main/kotlin/com/walletconnect/chat/engine/domain/ChatEngine.kt:238]: unregister won't have it in near future.

**Pull request merged**